src/errors.ts

```diff
@@ -34,7 +34,7 @@ export interface MatcherError extends Error {
  * Enumeration with all possible types for navigation failures. Can be passed to
  * {@link isNavigationFailure} to check for specific failures.
  */
-export enum NavigationFailureType {
+export const enum NavigationFailureType {
```

Comment:
Since this enum is exposed to the user, I doubt it's a good idea to make it const. It needs to be exposed to JS users as well